### PR TITLE
feat(registry): change the default registry from dockerhub to quay.io 

### PR DESCRIPTION
**Pull request by brokenpip3**
# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes tests (if functionality changed/added)
- [ ] Includes docs (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/jenkinsci/kubernetes-operator/blob/master/CONTRIBUTING.md#commit-messages)

_See [the contribution guide](https://github.com/jenkinsci/kubernetes-operator/blob/master/CONTRIBUTING.md) for more details._

## Reviewer Notes

Let's change the registry to avoid pull limitations and make the images name more near to the project name.
In the future we can improve the github ci to let it become less makefile dependent.
As soon a operator image will be pushed in quay.io I will create another PR to fix the doc with that reference.

# Release Notes

```
Changed the default registry from docker-hub to quay.io
```

### CODE REVIEW

File: .github/workflows/deploy-nightly.yaml [hunk 1]
```diff
@@ -65,11 +65,11 @@ jobs:
           eval $(bin/minikube docker-env)
           make helm-e2e E2E_TEST_ARGS='-ginkgo.v'
 
-      - name: Login to DockerHub
+      - name: Login to Quay.io
         uses: docker/login-action@v1
         with:
-          username: ${{ secrets.DOCKER_HUB_USERNAME }}
-          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+          username: ${{ secrets.QUAY_IO_USERNAME }}
```
Comment (prryb) [inline .github/workflows/deploy-nightly.yaml:71]:
```suggestion
          username: ${{ secrets.QUAYIO_USERNAME }}
```

File: .github/workflows/deploy-nightly.yaml [hunk 2]
```diff
@@ -65,11 +65,11 @@ jobs:
...
+          password: ${{ secrets.QUAY_IO_ACCESS_TOKEN }}
```
Comment (prryb) [inline .github/workflows/deploy-nightly.yaml:72]:
```suggestion
          password: ${{ secrets.QUAYIO_TOKEN }}
```

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -44,11 +44,11 @@ jobs:
         with:
           tag_name: ${{ env.VERSION }}
 
-      - name: Login to DockerHub
+      - name: Login to Quay.io
         uses: docker/login-action@v1
         with:
-          username: ${{ secrets.DOCKER_HUB_USERNAME }}
-          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+          username: ${{ secrets.QUAY_IO_USERNAME }}
```
Comment (prryb) [inline .github/workflows/release.yaml:50]:
```suggestion
          username: ${{ secrets.QUAYIO_USERNAME }}
```

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -44,11 +44,11 @@ jobs:
...
+          password: ${{ secrets.QUAY_IO_ACCESS_TOKEN }}
```
Comment (prryb) [inline .github/workflows/release.yaml:51]:
```suggestion
          password: ${{ secrets.QUAYIO_TOKEN }}
```

**Review by prryb — APPROVED** [review]
Hey, I've added those secrets before and forgot to let you know, figured out it's easier to just rename them here, rather than regenerate/add new.

**Comment by brokenpip3** [comment]
thanks @prryb :)

**Pull request merged**